source/math/FloatMatrix.ooc

```diff
@@ -381,9 +380,12 @@ FloatMatrix: cover {
 		submatrix
 	}
 	inverse: func -> This {
-		version(safe)
-			raise(!this take() isSquare, "Matrix must be square in FloatMatrix inverse")
-		result := this take() cofactors() transpose() / this determinant()
+		t := this take()
+		version(safe) {
+			raise(!t isSquare, "Matrix must be square in FloatMatrix inverse")
+			raise(t determinant() equals(0.f), "Matrix is singular in FloatMatrix inverse")
```

Comment:
Ah alright. It's common to think that way, but it's not a penalty. Since we have to do `t determinant()` later anyway, its value will need to be put *somewhere* on the stack, regardless of whether we do it explicitly (with variable assignment) or not.

Also, no we don't really need to care about optimizing the safe version, but on the other hand, determinant calculation can be costly and this seems simple enough to fix.